src/java.base/share/classes/java/util/zip/Deflater.java

```diff
@@ -49,29 +49,33 @@
  * thrown.
  * <p>
  * This class deflates sequences of bytes into ZLIB compressed data format.
- * The input byte sequence is provided in either byte array or byte buffer,
+ * The input byte sequence is provided in either byte array or {@link ByteBuffer},
  * via one of the {@code setInput()} methods. The output byte sequence is
- * written to the output byte array or byte buffer passed to the
+ * written to the output byte array or {@code ByteBuffer} passed to the
  * {@code deflate()} methods.
  * <p>
- * The following code fragment demonstrates a trivial compression
- * and decompression of a string using {@code Deflater} and
- * {@code Inflater}.
- * {@snippet id="compdecomp" lang="java" class="Snippets" region="DeflaterInflaterExample"}
+ * To release resources used by the {@code Deflater}, applications must close the
+ * {@code Deflater} by calling either the {@link #end()} or the {@link #close()} method.
+ * After the {@code Deflater} has been closed, subsequent calls to several methods
+ * of the {@code Deflater} will throw an {@link IllegalStateException}.
```

Comment:
This paragraph uses the definite article but there isn't a specific Deflater to speak of, and it's not a singleton. The first sentence of this paragraph might be better if re-worded "To release the resources used a Deflater, an application must close it by invoking its end() or close() method".

Reply:
I have reworded the sentence to follow your input. Addtionally, I've removed the second sentence, since as you note, relevant methods on the Inflater/Deflater (as part of this PR) already have been updated to state that they throw an `IllegalStateException` if the Inflater/Deflater is already closed.